src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  let parts = sourceString.split(';');
+
+  parts = parts.filter((part) => part.trim() !== '');
+
+  for (const part of parts) {
+    const partKeyValue = part.split(':');
+
+    for (let i = 0; i < partKeyValue.length; i++) {
+      const key = partKeyValue[0].trim();
+      const value = partKeyValue[1].trim();
+
+      result[key] = value;
```

Comment:
The nested 'for' loop is iterating over the 'partKeyValue' array but is using fixed indexes (0 and 1) inside the loop, which is redundant and could lead to incorrect behavior if 'partKeyValue' has more than two elements. Instead, directly access 'partKeyValue[0]' and 'partKeyValue[1]' outside of the loop.